autogpt_platform/backend/backend/data/graph.py

```diff
@@ -543,20 +548,38 @@ async def get_graph(
     where_clause: AgentGraphWhereInput = {
         "id": graph_id,
     }
+
     if version is not None:
         where_clause["version"] = version
-    elif not template:
-        where_clause["isActive"] = True
-
-    # TODO: Fix hack workaround to get adding store agents to work
-    if user_id is not None and not template:
-        where_clause["userId"] = user_id
 
     graph = await AgentGraph.prisma().find_first(
         where=where_clause,
         include=AGENT_GRAPH_INCLUDE,
         order={"version": "desc"},
     )
+
+    if not graph:
+        return None
+
+    if graph.userId == user_id:
+        return GraphModel.from_db(graph, for_export) if graph else None
+
+    # If the graph is not owned by the user, we need to check if it's a store listing.
+    if not version:
+        version = graph.version
+
+    store_listing_where: StoreListingWhereInput = {
+        "agentId": graph_id,
+        "agentVersion": version,
+    }
+
+    store_listing = await StoreListing.prisma().find_first(where=store_listing_where)
+
+    # If it is not a store listing, return None
+    if not store_listing:
+        return None
```

Comment:
What's this for? Doesn't this prevent users from viewing their own private agents?